pkg/parser/parser_test.go

```diff
@@ -0,0 +1,86 @@
+package parser
```

Comment:
```suggestion
package parser_test
```

We should use `_test` for tests that test the users api of our package, for unit tests we can use the `parser` package.

From my understanding of all the tests you've written, they are all integration tests and not unit tests, so I think `package parser_test` makes more sense for these tests.